target_chains/ethereum/entropy_sdk/solidity/PRNG.sol

```diff
@@ -0,0 +1,132 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+/// @title PRNG Library
+/// @notice A library for pseudorandom number generation and related utility functions
+/// @dev This PRNG library is designed to work with Pyth Entropy as the seed source.
+///      Pyth Entropy provides secure, rapid random number generation for blockchain applications,
+///      enabling responsive UX for NFT mints, games, and other use cases requiring randomness.
+library PRNG {
+    /// @notice Generate a random byte between 0 and 255
+    /// @param seed The Pyth Entropy seed (bytes32)
+    /// @return A random uint8 value
+    function randomByte(bytes32 seed) internal pure returns (uint8) {
```

Comment:
@baileynottingham can you think of a way to ensure that the user of this library doesn't accidentally pass the same Entropy seed to multiple of these random functions? E.g., the following code would be easy to write but also a security mistake:

```
bytes32 entropySeed = ... // get from entropy
unit8 rand = PRNG.randomByte(entropySeed);
uint256 rand2 = PRNG.randInt(entropySeed);
```

In most programming languages, a PRNG has internal state to track the current seed. Each time the user calls a function requesting randomness, the PRNG automatically updates that state.

Reply:
Thank you for your input. I have updated the code to be a contract, allowing it to store an internal seed and nonce. The `nextBytes32()` function now returns a value and updates the nonce. This function is utilized internally by all other functions, ensuring that the nonce is consistently updated. As a result, the following code works correctly and returns a different number each time:
`
PRNG prng = new PRNG(pythEntropySeed);
uint256 randomNumber1 = prng.randUint();
uint256 randomNumber2 = prng.randUint();
uint256 randomNumber3= prng.randUint();
uint256 randomNumber4 = prng.randUint();
`